### ISSUE TITLE
Add other unrelated materials to the appendix

### ISSUE THREAD

**Issue by RojerGS** [PierreSenellart/apxproof#44]
I can't seem to figure out how to add tables and etc. to my appendix while also using `apxproof` to add some proofs to the appendix. If I add an explicit `\appendix` to my `.tex` file, then the proofs that were automatically sent to the appendix will restart the appendix counter...

**Comment by PierreSenellart** [comment]
`\begin{toappendix}` and `\end{toappendix}` around the material you want sent to the appendix can be used for that purpose, and this allows keeping this extra material where it naturally fits within the paper. Let me know if this works for you.

**Comment by RojerGS** [comment]
> 
> 
> `\begin{toappendix}` and `\end{toappendix}` around the material you want sent to the appendix can be used for that purpose, and this allows keeping this extra material where it naturally fits within the paper. Let me know if this works for you.

Thanks for your reply. It works in the sense that it adds things to the appendix, but it triggers a funny bug: the things enclosed with `\begin{toappendix} \end{toappendix}` get appended to the end of the appendix, but `apxproof` generates an extra (empty) section of proofs corresponding to the section that preceded my `\begin{toappendix}` code.

This is my TeX:

```tex
% ...
\input{rtt_data}
\input{concl}                   % Conclusions section.
\begin{toappendix}
    \input{heuristic_results}
\end{toappendix}
```

and that produces the following:

![Screenshot_1](https://user-images.githubusercontent.com/5621605/122728497-1bd44b80-d270-11eb-8bd4-b766ad51988f.png)

If I move the `\begin{toappendix}` up, so that the code looks like

```tex
% ...
\input{rtt_data}
\begin{toappendix}
    \input{heuristic_results}
\end{toappendix}
\input{concl}                   % Conclusions section.
```

then the output becomes

![Screenshot_2](https://user-images.githubusercontent.com/5621605/122728608-37d7ed00-d270-11eb-9469-ebb5726547e2.png)

Neither section 4 nor 5 have any `apxproof`-related environments/commands.

**Comment by PierreSenellart** [comment]
This is working as expected, I think. Since the figures are part of a section in the text, a section title for this section appears in the appendix – it looks weird in this case because it is floating material, but it it were non-floating material (say, text), it would be desired to have a corresponding section in the appendix. There is a way to tell apxproof not to produce a section title for a specific section, that has appendix material. Just add `\nosectionappendix` after the corresponding '\section' command. Does this work for you? If not, if you provide a minimal complete example I can check it out.

**Comment by RojerGS** [comment]
I managed to fix my issue with one usage of `\nosectionappendix`, so thanks for that, but I don't think things are working as expected.

I created a minimal working example project on Overleaf: https://www.overleaf.com/3853347783qbqygjfzjywh
You can view and edit.
The project contains the main file with the following code:

```tex
\documentclass{article}
\usepackage[utf8]{inputenc}
\usepackage{apxproof} % Defer proofs to the appendix.

\begin{document}

\input{section}
\input{dummy_section}

\begin{toappendix}
    \input{tables}
\end{toappendix}

\end{document}
```

Each of the inputted files has a `\section` and a line of text:

```tex
% section.tex
\section{Section refers to tables}

I like tables. I have a table in \ref{sec:tables}
```

```tex
% dummy_section.tex
\section{Dummy section}

This has nothing relevant.
```

```tex
% tables.tex
\section{My tables}
\label{sec:tables}

\begin{table}[]
    \centering
    \begin{tabular}{c|c}
        1 & 2 
    \end{tabular}
    \caption{Caption}
    \label{tab:my_table}
\end{table}
```

This produces the following PDF:

![Screenshot_3](https://user-images.githubusercontent.com/5621605/122737139-a7ea7100-d278-11eb-8b64-05e9d0540148.png)


I don't see a reason why there should be an (empty) appendix section for the dummy section, as it contains _nothing_ relevant. As we move the `\begin{toappendix}\input{tables}\end{toappendix}` around, the (empty) appendix section that gets created changes.

**Comment by PierreSenellart** [comment]
Thanks. This is indeed not normal. Weirdly enough, the problem doesn't happen if the content of the `tables.tex` file is copied into the `main.tex` file instead of being `\input`ed. I will investigate this.

**Comment by RojerGS** [comment]
Good luck.
Just to be clear, I was able to work around this, so this issue doesn't affect me directly any more; feel free to close this issue (and open one specifically for the bug) if you want.

**Comment by PierreSenellart** [comment]
This required a somewhat ad-hoc fix, to detect a `\section` at the beginning of a file that is loaded with `\input`, but I have implemented this in bd56db3, and I am releasing v1.2.2 today that incorporates this fix.
